WebRTC Release r2.1 candidate for Fall25

stroncoso-quobis opened the pull request:
#### What type of PR is this?

Add one of the following kinds:
* subproject management

#### What this PR does / why we need it:

* Fall 2025 meta-release candidate

#### Which issue(s) this PR fixes:

Fixes #82 

#### Special notes for reviewers:

This PR includes all changes from main.

#### Changelog input

```
 release-note
Updated README with latest links
Updated CHANGELOG with release upgrades
- webrtc-registration
fix: Updated deviceId format, examples and tests
- webrtc-call-handling
fix: Added 400 error response
- webrtc-events
fix: Updated deviceId format, examples and tests
```

#### Additional documentation 

This section can be blank.

```
docs
Updated API-Readiness with Wiki link and ICM & COMMS new release
```

stroncoso-quobis commented on CHANGELOG.md:
```diff
@@ -3,6 +3,7 @@
 
 ## Table of contents
 
+- **[r1.3](#r13)**
```
Changes applied

stroncoso-quobis commented on CHANGELOG.md:
```diff
@@ -17,6 +18,66 @@ The below sections record the changes for each API version in each release as fo
 * for subsequent release-candidate(s), only the delta to the previous release-candidate
 * for a public release, the consolidated changes since the previous public release
 
+# r1.3
```
Changes applied

stroncoso-quobis commented on CHANGELOG.md:
```diff
@@ -17,6 +18,66 @@ The below sections record the changes for each API version in each release as fo
+
+* webrtc-registration v0.2.0
+* webrtc-call-handling v0.2.0
+* webrtc-events v0.1.0
```
API versions updated

stroncoso-quobis replied:
Also stated that this is a pre-release

stroncoso-quobis commented on README.md:
```diff
@@ -29,22 +29,22 @@ Sandbox API Repository to describe, develop, document, and test the WebRTC Servi
-  [[YAML]](https://github.com/camaraproject/WebRTC/blob/r1.2/code/API_definitions/webrtc-registration.yaml)
-  [[View it on ReDoc]](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/WebRTC/r1.2/code/API_definitions/webrtc-registration.yaml&nocors)
-  [[View it on Swagger Editor]](https://camaraproject.github.io/swagger-ui/?url=https://raw.githubusercontent.com/camaraproject/WebRTC/r1.2/code/API_definitions/webrtc-registration.yaml)
+  * **webrtc-registration v0.3.0**
```
Replaced with the corresponding -rc.1 suffix

stroncoso-quobis reviewed (commented):
Suggestions noted and reviewed:
- Included `-rc.1` suffix on API versions
- Renamed release to 2.x as it is a new releasing cycle
- Identified r2.1 as a pre-release, and separated from public release 1.3

Open to review.
Commonalities changes still pending to be approved and merged at #88 

Best regards,

stroncoso-quobis commented:
Hi,
- Merged main sources to convert this PR into an all-in-one release candidate.
- Version tags updated to align with releasing notes.

Best regards,

hdamker commented:
Two questions regarding /code/API_definitions/README.MD:
* is the page up-to-date? I can't "Notification Channel" in the repository
* wouldn't it be better to use relative links instead of absolute links into `main`? The benefit would be that people will stay within the context (fork, branch, release, ...)

hdamker commented:
@camaraproject/web-rtc_maintainers @camaraproject/web-rtc_codeowner Can anyone pick up the necessary todos for this PR? The goal to get this done at latest beginning of next week (July 27th). Afterwards most of Release Management team will be on vacation and we have to postpone WebRTC to Spring26.

stroncoso-quobis commented on code/API_definitions/webrtc-registration.yaml:
- Same as in other APIs: Server URL
  - Fixed
- Definition of deviceId: string defined, but not format limited (security risk)
  - Fixed, included same pattern as recommended for x-correlator

stroncoso-quobis commented on code/API_definitions/webrtc-registration.yaml:
- All test files reference v0.2.0 in feature line -> should match API version 0.3.0-rc.1
  - resource url fixed to v0.3rc1

- File Naming Inconsistency
  - Updated filenames and scenarios as the operationId using `registration` is more accurate with the API meaning and behavior

- Scenario Path Inconsistencies
  - Fixed

hdamker replied:
> Test Data Format
> 
> * createSession test uses deviceId: "1qazxsw23edc"
> * API description states deviceId should be in UUID format
> 
> Duplicate Scenario ID
> 
> * deleteSession.feature has @webrtc_registration_getSession_400.2_empty_request
> * Appears to be copy-paste error (should be deleteSession)

@stroncoso-quobis seems that these two issues are not fixed, right?

stroncoso-quobis commented on documentation/API_documentation/webrtc-call-handling-API-Readiness-Checklist.md:
```diff
@@ -1,12 +1,12 @@
 # API WebRTC Call Handling
 
-Checklist for webrtc-call-handling v0.2.0 in r1.2
+Checklist for webrtc-call-handling v0.3.0 in r2.1
```
Applied

stroncoso-quobis commented on documentation/API_documentation/webrtc-events-API-Readiness-Checklist.md:
```diff
@@ -1,6 +1,6 @@
 # API WebRTC events
 
-Checklist for webrtc-events v0.1.0 in r1.2
+Checklist for webrtc-events v0.2.0 in r2.1
```
Applied

stroncoso-quobis commented on documentation/API_documentation/webrtc-registration-API-Readiness-Checklist.md:
```diff
@@ -1,6 +1,6 @@
 # API WebRTC Registration
 
-Checklist for webrtc-registration v0.2.0 in r1.2
+Checklist for webrtc-registration v0.3.0 in r2.1
```
Applied

stroncoso-quobis reviewed (commented):
Fixed all comments, mostly simply path updates.

Really sorry with the webrtc-events API file, test and naming convention, improving on each release. Apologies in advance.

Commiting changes right away.

hdamker commented on code/API_definitions/webrtc-registration.yaml:
```diff
@@ -294,6 +294,8 @@ components:
       properties:
         deviceId:
           type: string
+          pattern: ^[a-zA-Z0-9-]{0,256}$
+          example: "7d444840-9dc0-11d1-b245-5ffdce74fad2"
           description: |-
             The deviceId of the client in UUID format. A unique identifier for
             the physical device where a registration is initiated. Generated
```
That pattern is unnecessarily complex as the API Provider can follow the UUID format, I would just recommend:

```suggestion
        deviceId:
          type: string
          format: uuid
          description: |-
            The deviceId of the client in UUID format. A unique identifier for
            the physical device where a registration is initiated. Generated
```

hdamker reviewed (commented):
The done changes are looking good to me, just one recommendation to use the predefined UUID.

hdamker commented on code/API_definitions/webrtc-call-handling.yaml:
For later consideration:

```
Address:
      type: string
      description: Subscriber address (Sender or Receiver)
      pattern: '^(tel:+[0-9]{5,15}|sip:.+@.+|.+)$'
      example: 'tel:+11234567899'
```

The final .+ alternative is very permissive and could accept invalid formats
Recommendation: Consider tightening the pattern or documenting what other formats are expected.

hdamker commented:
@stroncoso-quobis see comments above ... almost there 👍 

Do you plan to merge #93 and #94 before the rc.1?

stroncoso-quobis commented:
Hi @hdamker 

**Changes applied**

- Notification channel references removed from code/API_documentation/README.md
- Updated the webrtc-events tests to use the correct URL /webrtc-events
- Replaced the deviceId complex pattern with the simplier `uuid` as recommended. I have kept  the example for clarification purposes.

I left the comment regarding the address field as a bug  ( #95 ) as I think that it needs to be discussed with the team.

**Bugfixes at #93 and #94**

> _Do you plan to merge #93 and #94 before the rc.1?_

We intend is to approve them and try to include them in this meta-release. If my colleagues approve them, I will merge them into the main branch and then into this release branch.

I'm not really sure about the procedure and recommendations, but I see two options:
- Launch rc1: merge them after the main merge, re-review and publish) or
- Launch rc2: merge rc1 into main, then launch release rc2, making fixes wait as PRs

My personal preference is to launch them as rc1, as this involves less administrative work. It seems simpler to merge them here, validate and release. I am already in contact with my colleagues and expect them to approve the PRs within the next days.

Best regards,

hdamker commented on documentation/API_documentation/webrtc-events-API-Readiness-Checklist.md:
```diff
@@ -1,12 +1,12 @@
-|  1 | API definition                               |   M   |   M   |   M   |   M   |   Y   | [link](/code/API_definitions/webrtc-events.yaml) |
-|  2 | Design guidelines from Commonalities applied |   O   |   M   |   M   |   M   |   Y   | r2.3 |
-|  3 | Guidelines from ICM applied                  |   O   |   M   |   M   |   M   |   Y   | r2.3 |
+|  1 | API definition                               |   M   |   M   |   M   |   M   |   InProgress   | [link](/code/API_definitions/webrtc-events.yaml) |
+|  2 | Design guidelines from Commonalities applied |   O   |   M   |   M   |   M   |   InProgress   | r3.2 |
+|  3 | Guidelines from ICM applied                  |   O   |   M   |   M   |   M   |   InProgress   | r3.2 |
```
Please change InProgress before the release

hdamker commented on documentation/API_documentation/webrtc-registration-API-Readiness-Checklist.md:
```diff
@@ -1,12 +1,12 @@
-|  1 | API definition                               |   M   |   M   |   M   |   M   |   Y   | [link](/code/API_definitions/webrtc-registration.yaml) |
-|  2 | Design guidelines from Commonalities applied |   O   |   M   |   M   |   M   |   Y   | r2.3 |
-|  3 | Guidelines from ICM applied                  |   O   |   M   |   M   |   M   |   Y   | r2.3 |
+|  1 | API definition                               |   M   |   M   |   M   |   M   |   InProgress   | [link](/code/API_definitions/webrtc-registration.yaml) |
+|  2 | Design guidelines from Commonalities applied |   O   |   M   |   M   |   M   |   InProgress   | r3.2 |
+|  3 | Guidelines from ICM applied                  |   O   |   M   |   M   |   M   |   InProgress   | r3.2 |
```
Please change InProgress before the release

hdamker reviewed (commented):
I resolved all conversations which I have seen as addressed, only one with two small issues seem to be left.

BTW: I suppose it is time to change "InProgress" to "Y"

hdamker reviewed (approved):
Looks finally good to me from release management perspective.

Next actions to be completed for M3:

- PR merged (by API repository codeowner)
- Release created within GitHub (by API repository codeowner)
- Release Tracker updated (with API version, creation date of the release and the release tag link)

hdamker commented:
> We will de better on next releases.

@stroncoso-quobis There is a lot to be improved in the release process as well, lot of manual work and checks which should be automated. If you have some time have a look on https://github.com/camaraproject/ReleaseManagement/pull/275 and let me know what do you think - overdue or overkill?

hdamker commented on CHANGELOG.md:
```diff
@@ -17,6 +18,86 @@ The below sections record the changes for each API version in each release as fo
+* Provided a concrete example for the `registration-ends` event subscription.
+
+### Changes
+* fix: Updated deviceId format, examples and tests at #91
```
Seems that the links are not working in the resulting changelog in this format, propose to change them to the format which GitHub is creating within draft releases, then they will work also in local clones/downloads:

```suggestion
* fix: Updated deviceId format, examples and tests at https://github.com/camaraproject/WebRTC/pull/91
```

Please replace the other occurrences accordingly

hdamker reviewed (commented):
Ready to approve again ... but just seen that the CHANGELOG can be improved regarding the links. "#XX" format will be rendered in GitHub descriptions and comments, but as far as I know not in markdown documents.

stroncoso-quobis commented:
Hi @hdamker ,

> Now I'm little bit confused: I did my review around 18:00 UTC yesterday and your latest commits on your branch are from around 11:30 UTC

No clue at all... I was trying to investigate, but I can't reach any conclusion. It was 8PM on my timezone, it must be any kind of automation, no local traces at all on my side. Following that path, I look for the audit logs at GitHub, but I can not access to the admin section of the GitHub repo, so I can't figure exactly what happen and from where it happens. If you take a look there, maybe you can find something... Also I have 2FA activated and personal security audit logs does not show any activity... :thinking: intriguing... 

> Ready to approve again ... but just seen that the CHANGELOG can be improved regarding the links. "#XX" format will be rendered in GitHub descriptions and comments, but as far as I know not in markdown documents.

Fixed CHANGELOG at 8aad9d74907802560ccc836324b11926e7e4bd01

Re-reviewed markdown rendering and look for any. If it looks ok to you, please approve, and we will release.

Best regards,

hdamker reviewed (approved):
Another try to approve

hdamker reviewed (dismissed):
Next try to approve from Release Management (after removing branch protection on *release*)
